praw/models/reddit/flair.py

```diff
@@ -0,0 +1,256 @@
+"""Provide the Flair class."""
+from typing import Any, Dict, Optional, TypeVar, Union
+from .base import RedditBase
+
+_AdvancedSubmissionFlair = TypeVar("_AdvancedSubmissionFlair")
+_LinkFlair = TypeVar("_LinkFlair")
+_RedditorFlair = TypeVar("_RedditorFlair")
+_RichFlairBase = TypeVar("_RichFlairBase")
+Reddit = TypeVar("Reddit")
+Submission = TypeVar("Submission")
+Subreddit = TypeVar("Subreddit")
+
+
+class RichFlairBase(RedditBase):
+    """A base class for rich flairs.
+
+    Flairs of this type are obtained by either
+    :class:`.SubredditLinkFlairTemplates`
+    or :class:`.SubredditRedditorFlairTemplates`.
+
+    Subclasses:
+
+        * :class:`.AdvancedSubmissionFlair`
+        * :class:`.RedditorFlair`
+
+    **Typical Attributes**
+
+    This table describes attributes that typically belong to objects of this
+    class. Since attributes are dynamically provided (see
+    :ref:`determine-available-attributes-of-an-object`), there is not a
+    guarantee that these attributes will always be present, nor is this list
+    necessarily comprehensive.
+
+    ======================= ===================================================
+    Attribute               Description
+    ======================= ===================================================
+    ``allowable_content``   The type of content allowed in the flair (
+                            acceptable values are ``all``, ``emoji`` or
+                            ``text``).
+    ``background_color``    The background color of the flair. Values can be a
+                            color name (``dark`` or a 6-digit RGB code
+                            ``#ffffff``.)
+    ``css_class``           The css class of the flair.
+    ``id``                  The flair template id.
+    ``max_emojis``          The amount of :class:`.Emoji` that can go in the
+                            flair.
+    ``mod_only``            Whether or not the flair is mod only or not.
+    ``richtext``            A list containing data about any richtext elements,
+                            including any emojis, present in the flair.
+    ``text``                The text of the flair.
+    ``text_color``          The color of the text
+    ``text_editable``       Whether or not the flair text can be edited.
+    ``type``                The type of the flair (``text`` or ``richtext``.)
+    ======================= ===================================================
+    """
+
+    STR_FIELD = "text"
+
+    def __eq__(self, other: Union[str, _RichFlairBase]) -> bool:
+        """Check that two instances of the class are equal."""
+        if isinstance(other, str):
+            return str(self) == other
+        return (
```

Comment:
For the new way of consistency, we should only return a true/false value directly when the instances match (plus the string version), and rely on super for everything else.

Reply:
This was a copy-paste of old code.

Reply:
Can we update it for this code?